app/script/user/UserHandleGenerator.js

```diff
@@ -0,0 +1,115 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.user = z.user || {};
+
+(function() {
+  const MAX_HANDLE_LENGTH = 21;
+
+  const RANDOM_WORDS_1 = ['acid', 'agate', 'agile', 'amber', 'aqua', 'arid', 'avid', 'azure', 'baby', 'basic', 'beige', 'best', 'black', 'blond', 'blue', 'brass', 'brave', 'brown', 'busy', 'chief', 'chill', 'clear', 'cold', 'cool', 'coral', 'cosy', 'cozy', 'cream', 'curly', 'cyan', 'dear', 'dry', 'early', 'even', 'fancy', 'fast', 'fit', 'folk', 'gold', 'green', 'grey', 'happy', 'hazy', 'icy', 'iron', 'kind', 'large', 'lazy', 'lemon', 'light', 'lilac', 'lime', 'lord', 'lucid', 'mauve', 'melt', 'merry', 'mint', 'nice', 'noir', 'ochre', 'odd', 'olive', 'opal', 'peach', 'pearl', 'pink', 'plain', 'purple', 'quiet', 'rapid', 'red', 'rock', 'rose', 'ruby', 'rust', 'sand', 'sassy', 'shiny', 'shy', 'silly', 'slow', 'small', 'stone', 'sweet', 'swift', 'talc', 'tame', 'tiny', 'topaz', 'torn', 'total', 'vinyl', 'violet', 'warm', 'white', 'wise', 'witty', 'yellow', 'young', 'zinc'];
+  const RANDOM_WORDS_2 = ['alligator', 'alpaca', 'ant', 'antelope', 'asp', 'badger', 'bat', 'bear', 'bee', 'beetle', 'bird', 'bison', 'bobcat', 'buffalo', 'buzzard', 'camel', 'caribou', 'carp', 'cat', 'catfish', 'cheetah', 'clam', 'cobra', 'cod', 'condor', 'cow', 'coyote', 'crane', 'crayfish', 'cricket', 'crow', 'deer', 'dog', 'dolphin', 'donkey', 'dove', 'duck', 'eagle', 'eel', 'elk', 'falcon', 'ferret', 'finch', 'fly', 'fox', 'frog', 'gazelle', 'giraffe', 'gnu', 'goat', 'goose', 'gopher', 'grouse', 'gull', 'halibut', 'hamster', 'hare', 'hawk', 'heron', 'herring', 'horse', 'husky', 'impala', 'jackal', 'jaguar', 'kangaroo', 'koala', 'lemur', 'lion', 'lizard', 'llama', 'lobster', 'mackerel', 'mole', 'moose', 'moth', 'mouse', 'mule', 'mussel', 'newt', 'octopus', 'orca', 'ostrich', 'otter', 'owl', 'ox', 'oyster', 'panda', 'panther', 'parrot', 'pelican', 'penguin', 'pigeon', 'pike', 'pony', 'quail', 'rabbit', 'racoon', 'ram', 'raven', 'salmon', 'sardine', 'seal', 'shark', 'sheep', 'sloth', 'snail', 'snake', 'squid', 'sturgeon', 'swan', 'tiger', 'tilapia', 'toad', 'trout', 'tuna', 'turkey', 'turtle', 'walrus', 'wapiti', 'wasp', 'weasel', 'whale', 'wolf', 'wombat', 'yak', 'zebra'];
+
+  function get_random_word_combination() {
+    return `${z.util.ArrayUtil.random_element(RANDOM_WORDS_1)}${z.util.ArrayUtil.random_element(RANDOM_WORDS_2)}`;
+  }
+
+  /**
+   * Creates variations of the given handle by appending random digits.
+   * @param {string} handle - Input
+   * @param {number} number_of_variations - Number of variants that should be generated
+   * @returns {Array<string>} Handle variations
+   */
+  function generate_handle_variations(handle, number_of_variations = 5) {
+    return new Array(number_of_variations).map(function(value, index) {
```

Comment:
This is not going to work. Map will skip but return undefined in an array: http://2ality.com/2013/11/initializing-arrays.html

````
return _.range(number_of_variations).map(function(value) {
  return append_random_digits(handle.slice(0, MAX_HANDLE_LENGTH - value), value);
}
````